microsoft-edge/webview2/concepts/working-with-local-content.md

```diff
@@ -38,7 +53,39 @@ You must specify the full path of the file, for every resource.  For example:
 file:///C:/Users/username/Documents/GitHub/Demos/demo-to-do/index.html
 ```
 
-When specifying a file URL, the app navigates to a specific path, not a domain.  As a result, you can't use cross-origin resources for navigating to a file URL, because `document.origin` is null.
+
+###### Cross-origin resources
+
+When specifying a file URL, the app navigates to a specific path, not a domain.  As a result, you can't use cross-origin resources for navigating to a file URL, because `location.origin` is `file://` and cross-origin rules are different for file URLs.
+
+
+###### Origin-based DOM APIs
+
+Documents loaded via file URL has an origin that is unique to each file, just like in the browser. So APIs that require an origin such as `localStorage` or `indexedDB` will work, but different file URL documents are not considered "same-origin".
+
+
+###### DOM APIs requiring secure context
+
+Some newer browser features are limited to https URLs and are not available to file URLs. This includes webcam APIs, geolocation APIs, and notification APIs, among others. See [Secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) for more information.
```

Comment:
```suggestion
Some newer browser features are limited to https URLs and are not available to file URLs. This includes webcam APIs, geolocation APIs, and notification APIs. See [Secure contexts](https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts) for more information.
```
, among others -- sufficiently implied, though this paragraph could possibly need it more than where there's a phrase like "such as".